sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp

```diff
@@ -500,17 +518,59 @@ __esimd_svm_atomic1(__ESIMD_DNS::vector_type_t<uint64_t, N> addrs,
 {
   __ESIMD_DNS::vector_type_t<Ty, N> retv;
 
-  for (int i = 0; i < N; i++) {
-    if (pred[i]) {
-      Ty *p = reinterpret_cast<Ty *>(addrs[i]);
+  for (int AddrIdx = 0; AddrIdx < N; AddrIdx++) {
+    if (pred[AddrIdx] == 0) {
+      // Skip Output vector elements correpsonding to
+      // predicates whose value is zero
+      continue;
+    }
 
-      switch (Op) {
-      case __ESIMD_NS::atomic_op::add:
-        retv[i] = __ESIMD_DNS::atomic_add_fetch<Ty>(p, src0[i]);
-        break;
-      default:
-        __ESIMD_UNSUPPORTED_ON_HOST;
+    // Keeping original values for return
+    retv[AddrIdx] = *((Ty *)addrs[AddrIdx]);
```

Comment:
this is wrong (see above [comment](https://github.com/intel/llvm/pull/6661/files#r958778500))

Reply:
Done. Return values from `__atomic_*` functions are used.